src/qt/applicationbase/undomanager.cpp

```diff
@@ -174,23 +165,24 @@ class AutoSaver {
     std::thread saver_;
 };
 
-UndoManager::UndoManager(InviwoMainWindow* mainWindow)
-    : mainWindow_(mainWindow)
-    , manager_{mainWindow_->getInviwoApplication()->getWorkspaceManager()}
-    , refPath_{filesystem::findBasePath()}
-    , autoSaver_{std::make_unique<AutoSaver>(mainWindow_->getInviwoApplication())}
+UndoManager::UndoManager(InviwoApplication* app, QObject* mainWindow, int numRestoreFiles,
+                         int restoreFrequency)
+    : app_(app)
+    , manager_{app->getWorkspaceManager()}
+    , refPath_{filesystem::findBasePath()},
+      autoSaver_{std::make_unique<AutoSaver>(numRestoreFiles, restoreFrequency)}
 
 {
 
     QApplication::instance()->installEventFilter(this);
 
-    mainWindow_->getInviwoApplication()->getProcessorNetwork()->addObserver(this);
+    app->getProcessorNetwork()->addObserver(this);
 
-    undoAction_ = new QAction(QIcon(":/svgicons/undo.svg"), QAction::tr("&Undo"), mainWindow_);
+    undoAction_ = new QAction(QIcon(":/svgicons/undo.svg"), QAction::tr("&Undo"), mainWindow);
```

Comment:
It here a reason to parent them to the mainWindow and not just this?

Reply:
Have not tested. 
Could try and see

Reply:
Tested with this and could not see any difference